### ISSUE TITLE
too complex dipolar pathway input scheme

### ISSUE THREAD

**Issue by luisfabib** [luisfabib/DeerAnalysis2#48]
In ``dipolarkernel`` the dipolar pathway information (lambdas, timings, and harmonics) are passed as a single matrix with the following structure: 

```` matlab
pathwayinfo = [Lambda0 lambda1 lambda2 
                NaN    T1      T2
                NaN    n1      n2];
K = dipolarkernel(t,r,pathwayinfo)
````

There are some issues with this current system: 
- Having to pass ``NaN`` values as an input is counter-intuitive and a has a huge potential for errors from the user-perspective
- Due to the row/column differentiation in MATLAB, we are requiring the ``Lambdas``, ``Ts`` and ``ns`` to be passed as columns, concatenated as rows. Most common way for users to define vectors is as rows (space or colon instead of semicolon as separator).

While writing a simple script I found myself stuck at the second issue every time I defined a new kernel. 

My proposal for an alternative input scheme would use cell arrays instead of numerical ones: 
- No need for ``NaN`` values
- No issues with dimensionality of the input, everything can be handled internally 
- Extremely straightforward concatenation of the three variables. 
```` matlab
pathwayinfo = {Lambdas Ts ns};
K = dipolarkernel(t,r,pathwayinfo)
````

@stestoll what do you think? I could implement the changes myself if we agree on the design.

**Comment by stestoll** [comment]
I think  the current input syntax is adequate. The mental model is to list parameters for each contribution in turn, one per row.

E.g. for 5-pulse DEER, there is the non-modulated contribution `[Lambda0 NaN]`, then there is the main dipolar modulation `[lambda1 0]` and then there is the residual 4-pulse signal `[lambda2 T0]`. All together: 
```Matlab
pathwayinfo = [Lambda0 NaN; lambda1 0; lambda2 T0]
````
or 
```Matlab
pathwayinfo(1,:) = [Lambda0 NaN];       % unmodulated
pathwayinfo(2,:) = [lambda1 0];         % 5-pulse modulation
pathwayinfo(3,:) = [lambda2 T0];        % 4-pulse modulation
```
For each contribution, all the parameters are adjacent, and it is very easy to add or remove a contribution.

Grouping all the lambdas together, and separately all the T0s, makes it harder to add or remove contributions.

The only aspect that non-optimal is the use of `NaN` to indicate the absence of a refocusing point.

**Comment by luisfabib** [comment]
I agree that the mental model is more appropriate for this design. However, I am still afraid that requesting a numerical matrix will cause more difficulties due to the issues mentioned above. 

I like this syntax, since it perfectly reproduces the mental model and avoids the row/column issues
```` matlab
pathwayinfo(1,:) = [Lambda0 NaN];       % unmodulated
pathwayinfo(2,:) = [lambda1 0];         % 5-pulse modulation
pathwayinfo(3,:) = [lambda2 T0];        % 4-pulse modulation
```` 
so maybe a point in-between would be this 
 
```` matlab
pathwayinfo{1} = Lambda0;          % unmodulated
pathwayinfo{2} = [lambda1 0];      % 5-pulse modulation
pathwayinfo{3} = [lambda2 T0];     % 4-pulse modulation
```` 
which keeps the mental model but removes any of the issues present in the other one. Instead of putting all lambas and timings into separate cells, each cell is given to each pathway.

**Comment by stestoll** [comment]
My experience with EasySpin users is that they struggle with the concept of cell arrays, so they are best avoided for usability reasons.

**Comment by luisfabib** [comment]
Ok, we'll go with the voice of experience. However if we keep this design we have to do a couple of things: 
- adapt the current documentation for ``dipolarkernel`` so that we write out the syntax as above. This way users can (as they often do) copy-paste that and adapt for their use. 
-  add an additional input parsening check that controls the orientation of the matrix. 

    Right now using 
    ````matlab
   K = dipolarkernel(t,r,[lambdas; [NaN Ts]].',Bmodel) 
    ````
    works correctly as expected, but transposing the matrix
    ````matlab
   K = dipolarkernel(t,r,[lambdas; [NaN Ts]],Bmodel) 
    ````
   returns a matrix of ``NaN`` instead of raising an error.

**Comment by stestoll** [comment]
Fixed with 8661c88160bbf0ea619816c5b7b3e08af6585655.
